htmd/smallmol/smallmol.py

```diff
@@ -1,105 +1,654 @@
+        chirals =  []
+        hybridizations = []
+        neighbors = []
+        bondtypes = []
+        chiraltags = []
+
+        _mol = self._mol
+
+        if smallMol is not None:
+            for f, v in smallMol.__dict__.items():
+                self.__dict__[f] = v
+            return
+
+        # Development note explicitHs removed
+        for a in _mol.GetAtoms():
+            i = a.GetIdx()
+            e = a.GetSymbol()
+            chiraltags.append(a.GetChiralTag())
+            idxs.append(i)
+            atomnames.append('{}{}'.format(e,i))
+            formalcharges.append(a.GetFormalCharge())
+            elements.append(e)
+            neighbors.append([na.GetIdx() for na in a.GetNeighbors()])
+            hybridizations.append(int(a.GetHybridization()))
+            bondtypes.append([int(b.GetBondType()) for b in a.GetBonds()])
+            if a.HasProp('_TriposPartialCharge'):
+                charges.append(a.GetPropsAsDict()['_TriposPartialCharge'])
+            else:
+                charges.append(0.000)
+            if a.HasProp('_CIPCode'):
+                chirals.append(a.GetPropsAsDict()['_CIPCode'])
             else:
-                self._mol = Chem.MolFromSmiles(mol)
+                chirals.append('')
 
-        # Don't feed garbage!
+        if _mol.HasProp('_Name'):
+            self.__dict__['ligname'] = _mol.GetProp('_Name')
         else:
-            raise ValueError("Unkown file type: '{}'.".format(type(mol)))
+            self.__dict__['ligname'] = 'UNK'
+
+        for k, v in _mol.GetPropsAsDict().items():
+            self.setProp(k,v, overwrite=True)
+
+        # Development note explicitHs removed
+        self.__dict__['totalcharge'] = sum(formalcharges)
+        self.__dict__['idx'] = np.array(idxs)
+        self.__dict__['atomname'] = np.array(atomnames)
+        self.__dict__['charge'] = np.array(charges )
+        self.__dict__['formalcharge'] = np.array(formalcharges)
+        self.__dict__['element'] = np.array(elements)
+        self.__dict__['chiral'] = np.array(chirals)
+        self.__dict__['hybridization'] = np.array(hybridizations)
+        self.__dict__['neighbors'] = np.array(neighbors)
+        self.__dict__['bondtypes'] = np.array(bondtypes)
+        self.__dict__['_chiraltags'] = np.array(chiraltags)
+
+        if _mol.GetNumConformers() != 0:
+            coords = _mol.GetConformer(0).GetPositions()
+            self.__dict__['coords'] =  coords[:,:,np.newaxis]
+        else:
+            self.__dict__['coords'] = np.zeros((self.numAtoms,3,1))
 
-        # Add hydrogens
-        if addHs:
-            self._mol = Chem.RemoveHs(self._mol)
-            self._mol = Chem.AddHs(self._mol, addCoords=True)
+    def copy(self):
+        """
+        Create a copy of the molecule object
+
+        Returns
+        -------
+        newsmallmol : :class:`SmallMol`
+            A copy of the object
+        """
+        newsmallmol = deepcopy(self)
+        return newsmallmol
```

Comment:
By the way, have you ever checked if this copies the `_mol` object? I am a bit worried about `rdKit` objects.

Reply:
The rdkit mol is copied correctly. The properties stored in the molecule and atoms are probably lost. All the important infomration are already stored in the __dict__, thus loosing that info are not anymore a problem.  Moreover, as I commented above, the '_mol' is just for debugging purpose